packages/runtime-common/realm-auth-client.ts

```diff
@@ -0,0 +1,111 @@
+import { MatrixClient } from './matrix-client';
+
+export class RealmAuthClient {
+  private realmURL: URL;
+  private jwt: string | undefined;
+  private matrixClient: MatrixClient;
+
+  constructor(
+    matrixUsername: string,
+    matrixPassword: string,
+    matrixURL: URL,
+    realmURL: URL,
+  ) {
+    this.realmURL = realmURL;
+    this.matrixClient = new MatrixClient(
+      matrixURL,
+      matrixUsername,
+      matrixPassword,
+    );
+  }
+
+  async getJWT() {
+    let extraPeriodBeforeExpirySeconds = 60;
```

Comment:
Perhaps `tokenRefreshLeadTimeSeconds` would be a better name

Reply:
Agree, less awkward